### PR TITLE
Port IMX platforms to use generic RAM layout

### PR DESCRIPTION
**Pull request by jordanrh1**
Port IMX platforms to use `mm/generic_ram_layout.h`. #2348 should be merged before this to ensure memory layout does not change.

### CODE REVIEW

**Comment by jordanrh1** [comment]
@MrVan This touches IMX.

File: core/arch/arm/plat-imx/conf.mk [hunk 1]
```diff
@@ -83,12 +89,33 @@ CFG_BOOT_SECONDARY_REQUEST ?= y
...
+ifeq ($(filter y, $(CFG_MX6Q) $(CFG_MX6D) $(CFG_MX6DL) $(CFG_MX6S)), y)
+CFG_TZDRAM_START ?= 0x4E000000
+CFG_TZDRAM_SIZE ?= 0x01F00000
+CFG_SHMEM_START ?= ($(CFG_TZDRAM_START) + $(CFG_TZDRAM_SIZE))
+CFG_SHMEM_SIZE ?= 0x00100000
+endif
+
+# i.MX6 SoloX specific config
+ifeq ($(filter y, $(CFG_MX6SX)), y)
+CFG_TZDRAM_START ?= (0x80000000 + $(CFG_DDR_SIZE) - 0x02000000)
```
Comment (MrVan) [inline core/arch/arm/plat-imx/conf.mk]: This needs to be `0x80000000 - 0x02000000 + $(CFG_DDR_SIZE)`, CFG_DDR_SIZE sometimes is 2GB, so first subtract the shmem size will avoid errors.

File: core/arch/arm/plat-imx/conf.mk [hunk 2]
```diff
@@ -83,12 +89,33 @@ CFG_BOOT_SECONDARY_REQUEST ?= y
...
+endif
+
 ifeq ($(filter y, $(CFG_MX7)), y)
 include core/arch/arm/cpu/cortex-a7.mk
 
 $(call force,CFG_SECURE_TIME_SOURCE_REE,y)
 CFG_BOOT_SECONDARY_REQUEST ?= y
 CFG_INIT_CNTVOFF ?= y
+
+CFG_TZDRAM_START ?= (0x80000000 + $(CFG_DDR_SIZE) - 0x02000000)
```
Comment (MrVan) [inline core/arch/arm/plat-imx/conf.mk]: Ditto

File: core/arch/arm/plat-imx/conf.mk (hunk 1 above)
Comment (MrVan) [reply core/arch/arm/plat-imx/conf.mk]: I met compile issue before. Seems no issue now.

**Comment by MrVan** [comment]
`Reviewed-by: Peng Fan <peng.fan@nxp.com>`

**Comment by jordanrh1** [comment]
Added Reviewed-by tags.

File: core/arch/arm/plat-imx/conf.mk (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/plat-imx/conf.mk]: Might depends on compilation toolchain.
I would recommend `CFG_TZDRAM_START ?= (0x80000000 - 0x02000000 + $(CFG_DDR_SIZE))`, but i can't tell its bulletproof.

**Comment by jforissier** [comment]
Nice cleanup, thanks.

**Pull request merged**